project/geosource/views.py

```diff
@@ -43,6 +43,14 @@ def refresh(self, request, pk):
         refresh_job = source.run_async_method("refresh_data", force=force_refresh)
         if refresh_job:
             source.status = Source.Status.PENDING.value
+            if source.report:
+                source.report.reset()
+                source.report.status = SourceReporting.Status.PENDING.value
```

Comment:
why don'y set PENDING in reset method ?

Reply:
I could set it in the reset method.  At first, I thought of the `reset` as the report's initial, pristine state. Therefore the status is set as `None`.

I'll change it to `PENDING`.

Reply:
Done.